sklearn/isotonic.py

```diff
@@ -235,6 +236,7 @@ def _build_f(self, X, y):
     def _build_y(self, X, y, sample_weight, trim_duplicates=True):
         """Build the y_ IsotonicRegression."""
         self._check_fit_data(X, y, sample_weight)
+        X = X.flatten()
```

Comment:
As suggested by the [numpy docs](https://numpy.org/devdocs/reference/generated/numpy.ravel.html) `reshape` is more likely to return a view:

```suggestion
        X = X.reshape(-1)
```